pyomo/contrib/mindtpy/nlp_solve.py

```diff
@@ -146,6 +151,7 @@ def handle_nlp_subproblem_tc(fixed_nlp, result, solve_data, config, cb_opt=None)
 
 
 def handle_subproblem_optimal(fixed_nlp, solve_data, config, cb_opt=None, fp=False):
+    # TODO: move this to algorithm base class.
```

Comment:
Yes

Reply:
We already have a `handle_subproblem_optimal` method in the algorithm base class. However, we still need this in the single tree implementation.